extensions/coap/COAPLoader.cpp

```diff
@@ -16,9 +16,36 @@
  * limitations under the License.
  */
 #include "core/FlowConfiguration.h"
+#include "core/logging/LoggerConfiguration.h"
 #include "COAPLoader.h"
 
+#ifdef WIN32
+#include <winsock2.h>
+#endif
+
 bool COAPObjectFactory::added = core::FlowConfiguration::add_static_func("createCOAPFactory");
+
+bool COAPObjectFactoryInitializer::initialize() {
+#ifdef WIN32
+  static WSADATA s_wsaData;
+  int iWinSockInitResult = WSAStartup(MAKEWORD(2, 2), &s_wsaData);
+  if (iWinSockInitResult != 0) {
+    logging::LoggerFactory<COAPObjectFactoryInitializer>::getLogger()->log_error("WSAStartup failed with error %d", iWinSockInitResult);
+    return false;
+  } else {
```

Comment:
Be  removing the else (which is fine because of the return above) we can also remove the else part of the ifdef, but it's just fancy, this should work fine.

Reply:
I think this is better readable this way.